Firestore/core/src/firebase/firestore/local/serializer.cc

```diff
@@ -140,6 +140,24 @@ std::unique_ptr<model::MaybeDocument> Serializer::DecodeMaybeDocument(
   }
 }
 
+void Serializer::EncodeDocument(Writer* writer, const model::Document& doc) const {
+  // Encode Document.name
+  writer->WriteTag({PB_WT_STRING, google_firestore_v1beta1_Document_name_tag});
+  writer->WriteString(rpc_serializer_.EncodeKey(doc.key()));
+
+  // Encode Document.fields (unless it's empty)
```

Comment:
Question: just curious, can the check for emptiness move to `EncodeObjectMap`, or is there a reason to keep it outside?

Reply:
Hm. I think that could be moved inside without any troubles. I'll address this in a followup (since this should apply within the remote serializer too, and I've got existing followups that rework this code a bit already.)